ratis-netty/src/main/java/org/apache/ratis/netty/server/DataStreamManagement.java

```diff
@@ -145,6 +150,22 @@ LocalStream getLocal() {
     public String toString() {
       return JavaUtils.getClassSimpleName(getClass()) + ":" + request;
     }
+
+    @Override
+    public List<RaftPeerId> getSuccessors(RaftPeerId peerId) throws IOException {
+      Map<RaftPeerId, List<RaftPeerId>> routingTable = request.getRoutingTable();
+      if (routingTable.containsKey(peerId)) {
+        return routingTable.get(peerId);
+      } else if (isPrimary()) {
+        // TODO(runzhiwang): if filestore support route table, we can remove this branch
```

Comment:
We should keep this branch as the default for the applications which does not support RoutingTable.